validphys2/src/validphys/core.py

```diff
@@ -215,25 +216,37 @@ def get_members(self):
 
 
 class CommonDataSpec(TupleComp):
-    def __init__(self, datafile, sysfile, plotfiles, name=None, metadata=None):
+    def __init__(self, datafile, sysfile, plotfiles, name=None, metadata=None, legacy=True):
+        self.legacy = legacy
+        self._metadata = metadata
         self.datafile = datafile
         self.sysfile = sysfile
-        self.plotfiles = tuple(plotfiles)
-        self._name = name
-        self._metadata = metadata
-        super().__init__(datafile, sysfile, self.plotfiles)
+        if legacy:
+            self.plotfiles = tuple(plotfiles)
+            super().__init__(datafile, sysfile, self.plotfiles)
+        else:
+            self.plotfiles = False
+            super().__init__(name)
 
     @property
     def name(self):
         return self.metadata.name
 
     @property
     def nsys(self):
-        return self.metadata.nsys
+        if self.legacy:
+            return self.metadata.nsys
+        else:
+            cd = self.load()
+            return cd.nsys
```

Comment:
Hi @scarlehoff, I think that loading a Commondata instance within CommondataSpec goes against the idea of CommondataSpec.

The reason is that CommondataSpec has a production rule, `produce_commondata`, that its being loaded during `reportengine`'s "compilation". As such it should only contain metadata and not actual data.

There might have been already a discussion about it, so sorry if this has been already discussed, but isn't it possible to add `nsys` to metadata.yaml as well?
I mean one can do it after having generated the artificial systematics with `filter.py`

Reply:
There's no way around this so I'm going to mark it as resolved.

Hopefully there are no many utilities using nsys without having loaded the data first so it should have no effect.